src/fal/cli/selectors.py

```diff
@@ -80,15 +93,45 @@ class SelectorPlan:
     unique_ids: List[str]
     children: bool
     parents: bool
+    type: SelectType
 
-    def __init__(self, selector: str, unique_ids: List[str], project_name):
+    def __init__(
+        self, selector: str, unique_ids: List[str], project_name, fal_dbt: FalDbt
+    ):
         self.children = _needs_children(selector)
         self.parents = _need_parents(selector)
+        self.type = _to_select_type(selector)
         node_name = _remove_graph_selectors(selector)
-        if _is_script_node(node_name):
+
+        if self.type == SelectType.MODEL:
+            self.unique_ids = [f"model.{project_name}.{node_name}"]
+        elif self.type == SelectType.SCRIPT:
             self.unique_ids = _expand_script(node_name, unique_ids)
+        elif self.type == SelectType.COMPLEX:
+            self.unique_ids = unique_ids_from_complex_selector(selector, fal_dbt)
+
+
+def unique_ids_from_complex_selector(select, fal_dbt: FalDbt) -> List[str]:
```

Comment:
this does not handle stuff like `tag:x+`, right? It would ignore scripts associated to those.

What if we do a recursive solution? we pass the resulting models we got from `tag:x` (let's say models `[a, c, e]`) to `SelectorPlan` again as `a+ c+ e+`.

Meaning: we process colons in general IN dbt and then process the resulting models as normal models we di here.

Reply:
Maybe it does, I am a little bit lost, to be honest.

Reply:
it does.. I added tests for it

Reply:
see the selector logic we parse if it requires parents or children so same thing applies to the complex tags